src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -139,6 +135,7 @@
      * @param maxRecordsInRam how many records to accumulate before spilling to disk
      * @param tmpDir Where to write files of records that will not fit in RAM
      */
+    @SuppressWarnings("unchecked")
```

Comment:
It is unclear to me why this code doesn't use `ArrayList` for `ramRecords`, as there is very little overhead if you preallocate at construction time. That would the remove the need for this annotation and the bogus `componentType` constructor argument.

If the API can't be changed at this point it would at least be better to narrow the scope of the `SuppressWarnings` annotation. This can be done by introducing a local variable, e.g.
```
        @SuppressWarnings("unchecked")
        T[] ramRecords = (T[]) Array.newInstance(componentType, maxRecordsInRam);
        this.ramRecords = ramRecords;
```

Reply:
I suspect that at the time of creation this was not the case. I think it is ok to change the API over time as long as we have sufficient time with deprecated constructors. Do you think that the benefits of using an ArrayList are worth that change?